c_glib/test/dataset/test-file-system-dataset.rb

```diff
@@ -56,6 +56,24 @@ def test_partitioning
   end
 
   def test_read_write
+    dataset, expected_table = create_dataset
+    assert_equal(expected_table, dataset.to_table)
+  end
+
+  def test_record_batch_reader
+    dataset, expected_table = create_dataset
+    reader = dataset.to_reader
+    record_batches = []
+    loop do
+      batch = reader.read_next
+      break if batch.nil?
+      record_batches << batch
+    end
+    table_from_batches = Arrow::Table.new(reader.schema, record_batches)
+    assert_equal(expected_table, table_from_batches)
```

Comment:
```suggestion
    assert_equal(expected_table, reader.read_all)
```

Reply:
That's much simpler thanks, not sure how I missed that :smile: